src/components/Balance.tsx

```diff
@@ -8,6 +8,7 @@ interface BalanceProps extends TextProps {
   unit?: string
   isDisabled?: boolean
   prefix?: string
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void
```

Comment:
I assume this is necessary but it shouldn't be. `onClick` should be inherited from `TextProps`

Reply:
I think I tried to pass it just like that but TS couldn't find it... will try again, maybe I did something wrong last time 🤔

Reply:
I think if you didn't pass it at all within the `Balance` component Cheems, it may just work? 

`<Text>` accepts `{...props}` and then the `TextProps` should allow an onClick event?

Reply:
update on that - yea, TextProps doesn't allow onClick at all, needs to be explicitly defined here. TextProps inherits only from SpaceProps, TypographyProps, LayoutProps, and they don't contain onClick definition